tests/e2e/specs/upe-split/shopper/shopper-deferred-intent-creation-upe-enabled.spec.js

```diff
@@ -102,6 +102,39 @@ describe( 'Enabled UPE with deferred intent creation', () => {
 			await shopperWCP.deleteSavedPaymentMethod( card.label );
 			await expect( page ).toMatch( 'Payment method deleted' );
 		} );
+
+		it( 'should show terms when checking the save checkbox', async () => {
+			await setupProductCheckout(
+				config.get( 'addresses.customer.billing' )
+			);
+			await fillCardDetails( page, card );
+			await shopperWCP.toggleSavePaymentMethod();
+
+			const frameHandle = await page.waitForSelector(
+				'#payment .payment_method_woocommerce_payments .wcpay-upe-element iframe'
+			);
+			const stripeFrame = await frameHandle.contentFrame();
+
+			const textElement = await stripeFrame.$( '.TermsText' );
+
+			const text = textElement.evaluate( ( el ) => el.innerText );
+
+			expect( text ).toContain(
+				'By providing your card information, you allow'
+			);
```

Comment:
I think `toContain()` matches items in an array. Using regex on the substring might be better here. 
```suggestion
			expect( text ).toMatch(
				/By providing your card information, you allow/
			);
```